semgrep/semgrep/cli.py

```diff
@@ -282,6 +290,35 @@ def cli() -> None:
         error_on_findings=args.error,
     )
 
+    if not args.disable_version_check:
+        quick_timeout = 2  # Don't block user's too long
+        endpoint = os.environ.get(
+            "VERSION_CHECK_URL", "https://semgrep.live/api/check-version"
+        )
+        try:
+            resp = requests.get(
+                endpoint,
+                headers={"User-Agent": f"Semgrep/{__VERSION__}"},
+                timeout=quick_timeout,
+            )
+        except Exception as e:
+            debug_print(f"Could not connect to version check URL: {e}")
+        else:
+            if resp.status_code != requests.codes.OK:
+                debug_print(
+                    f"Received HTTP error code from version check URL: {resp.status_code}"
+                )
+            try:
+                resp_json = resp.json()
+            except ValueError:
+                debug_print(f"Could not decode JSON object from version check URL.")
+            else:
+                latest_version = resp_json["latest"]
+                if latest_version > __VERSION__:
+                    print_msg(
+                        "A new version of Semgrep is available. Please see https://github.com/returntocorp/semgrep#upgrading for more information."
```

Comment:
Still have to add this header, but it'll include the relevant information for each install mechanism.